includes/admin/reports/class-wc-admin-report-v2.php

```diff
@@ -0,0 +1,120 @@
+<?php
+/**
+ * V2 report functionality
+ *
+ * @package WooCommerce\Admin\Reports
+ */
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit; // Exit if accessed directly.
+}
+
+if ( ! class_exists( 'WC_Admin_Report' ) ) {
+	require_once dirname( __FILE__ ) . '/class-wc-admin-report.php';
+}
+
+/**
+ * Class WC_Admin_Report_V2
+ * Makes use of the WC Admin plugin endpoints if it is not disabled.
+ */
+class WC_Admin_Report_V2 extends WC_Admin_Report {
+
+	/**
+	 * Overwrites the original sparkline to use the new reports data if WooAdmin is enabled.
+	 * Prepares a sparkline to show sales in the last X days.
+	 *
+	 * @param  int    $id ID of the product to show. Blank to get all orders.
+	 * @param  int    $days Days of stats to get.
+	 * @param  string $type Type of sparkline to get. Ignored if ID is not set.
+	 * @return string
+	 */
+	public function sales_sparkline( $id = '', $days = 7, $type = 'sales' ) {
+		$is_wc_admin_disabled = apply_filters( 'woocommerce_admin_disabled', false );
+		if ( $is_wc_admin_disabled ) {
+			return parent::sales_sparkline( $id, $days, $type );
+		}
+		$sales_endpoint = '/wc-analytics/reports/revenue/stats';
+		$start_date     = gmdate( 'Y-m-d 00:00:00', current_time( 'timestamp' ) - ( ( $days - 1 ) * DAY_IN_SECONDS ) );
+		$end_date       = gmdate( 'Y-m-d 23:59:59', current_time( 'timestamp' ) );
+		$meta_key       = 'net_revenue';
+		$params         = array(
+			'order'    => 'asc',
+			'interval' => 'day',
+			'per_page' => 100,
+			'before'   => $end_date,
+			'after'    => $start_date,
+		);
+		if ( $id ) {
+			$sales_endpoint     = '/wc-analytics/reports/products/stats';
+			$meta_key           = ( 'sales' === $type ) ? 'net_revenue' : 'items_sold';
+			$params['products'] = $id;
+		}
+		$request          = new \WP_REST_Request( 'GET', $sales_endpoint );
+		$params['fields'] = array( $meta_key );
+		$request->set_query_params( $params );
+
+		$response = rest_do_request( $request );
```

Comment:
Make use of `rest_do_request` our performance indicator api does the same thing internally, though this was the easiest step towards migrating this.